File: billing-v3/src/main/java/io/appmetrica/analytics/billingv3/impl/ProductTypeParser.java
package io.appmetrica.analytics.billingv3.impl;

import androidx.annotation.NonNull;
import com.android.billingclient.api.BillingClient;
import io.appmetrica.analytics.billinginterface.internal.ProductType;

public class ProductTypeParser {

    @NonNull
    public static ProductType parse(@NonNull final String type) {
        if (BillingClient.SkuType.INAPP.equals(type)) {
            return ProductType.INAPP;
        }
        if (BillingClient.SkuType.SUBS.equals(type)) {
            return ProductType.SUBS;
        }
        return ProductType.UNKNOWN;
    }
}

File: billing-v4/src/main/java/io/appmetrica/analytics/billingv4/impl/ProductTypeParser.kt
package io.appmetrica.analytics.billingv4.impl

import com.android.billingclient.api.BillingClient
import io.appmetrica.analytics.billinginterface.internal.ProductType

object ProductTypeParser {

    fun parse(type: String): ProductType {
        return when (type) {
            BillingClient.SkuType.INAPP -> ProductType.INAPP
            BillingClient.SkuType.SUBS -> ProductType.SUBS
            else -> ProductType.UNKNOWN
        }
    }
}
